projects/plugins/jetpack/modules/masterbar.php

```diff
@@ -22,11 +22,7 @@
 require __DIR__ . '/masterbar/admin-color-schemes/class-admin-color-schemes.php';
 require __DIR__ . '/masterbar/inline-help/class-inline-help.php';
 
-$is_proxied = isset( $_SERVER['A8C_PROXIED_REQUEST'] )
-			? sanitize_text_field( wp_unslash( $_SERVER['A8C_PROXIED_REQUEST'] ) )
-			: defined( 'A8C_PROXIED_REQUEST' ) && A8C_PROXIED_REQUEST;
-
-$should_use_nav_redesign = apply_filters( 'is_nav_redesign_enabled', $is_proxied && get_option( 'wpcom_admin_interface' ) === 'wp-admin' );
+$should_use_nav_redesign = function_exists( 'wpcom_is_nav_redesign_enabled' ) && wpcom_is_nav_redesign_enabled();
```

Comment:
The other checks in Jetpack already used this style. This is the only part place where we still use the filter.

Reply:
Re: [other instance](https://github.com/search?q=repo%3AAutomattic%2Fjetpack+function_exists%28+%27wpcom_is_nav_redesign_enabled%27+%29&type=code)

I see one of those is in jetpack-mu which can have a much quicker deploy timeline than jetpack. 

And the other is in the blaze package, im not sure how the release cycle works for these. But assume its probably more controllable than the jetpack plugin as well?

If thats the case, I say leave the filters and change the checks elsewhere to use them as well.

Reply:
Actually im realizing converging these is not a blocker for rollout. [Doing this ](https://github.com/Automattic/wpcomsh/pull/1714#discussion_r1501686861) on your wpcomsh side should allow the logic to be delivered even while this change is pending jetpack release.

Reply:
The other instances that you linked above already don't use the filter, while the one that I changed here already uses the correct default value for the filter (the same logic as `wpcom_is_nav_redesign_enabled()`.

In wpcomsh, that function is already defined even before the PR.

So, I don't think the wpcomsh PR needs any changes? They will work together even when we remove the apply_filter usage, in any deployment order of Jetpack and wpcomsh.

Reply:
Ooh... I'm starting to understand what you mean.

We're afraid that Jetpack (/ -mu-wpcom) is not already deployed with this change, when the time comes where we want to actually turn the toggle on?

In that case, you're right we still need to add the wpcomsh function to the filter to be 100% sure. 👍 😄

Reply:
Fair enough.

> while the one that I changed here already uses the correct default value for the filter

Just thinking maybe we will want to remove the proxy check before this change deploys with the next jetpack release or something? in which case we just have to call `add_filter` on the wpcomsh side and this side will pick up the changes from there removing the proxy. 

> So, I don't think the wpcomsh PR needs any changes?

Perhaps not. You don't necessarily need to change anything there now, but if the above ends up being the case we have the option to add it to hook into this side.

Reply:
🤣 somehow your most recent comment didnt render before my last reply 😕 

anyways, it sounds like you get what I was saying 😆